src/Kongverge/Program.cs

```diff
@@ -135,6 +144,24 @@ public void Apply(IServiceProvider serviceProvider)
                 {
                     settings.InputFolder = InputFolder.Value();
                 }
+
+                if (BasicAuthPassword.HasValue())
+                {
+                    if (BasicAuthPassword.Value() == null)
```

Comment:
This logic where the password is fetched from either of two sources should be done as part of the validation step above (and then the result cached for when we need to apply it here).

Reply:
The validation steps all exit if there is an invalid configuration, where the apply method is used to read the values into the configuration.

Reply:
You haven't understood my comment then. In order to validate completely, you need to check both sources for the password. That means this code has to go into the validate step. But then you don't want to have to do the "checking both sources" again, so you just save the password into a field that you can use here directly.